src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,354 @@
+    }
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
+    // Start at 1 because index 0 is for "M"
+    for (size_t i = 1; i < colliders.size(); i++){
+        if (!colliders[i].hasKey("collider")) {
+                throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                    "Incorrect YAML input for LMR-R reaction. Please review implementation guide.");
+        }
+        if (!colliders[i].hasKey(eig_eps_key)) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "All collider strengths must be defined uniformly as either eig0 or eps. No mixing and matching is allowed.");
+        }
+        // Save data to colliderInfo, which will make it accessible by getParameters
+        colliderInfo.insert({colliders[i]["collider"].as<string>(), colliders[i]});
+        colliderNames.push_back(colliders[i]["collider"].as<string>());
+        ArrheniusRate epsObj_i;
+        // eig0 and eps are ONLY interchangeable due to the requirement that eps_M have parameters {A: 1, b: 0, Ea: 0}
+        params["A"] = colliders[i][eig_eps_key]["A"].asDouble() / colliders[0][eig_eps_key]["A"].asDouble();
+        params["b"] = colliders[i][eig_eps_key]["b"].asDouble() - colliders[0][eig_eps_key]["b"].asDouble();
+        params["Ea"] = colliders[i][eig_eps_key]["Ea"].asDouble() - colliders[0][eig_eps_key]["Ea"].asDouble();
+        epsObj_i = ArrheniusRate(AnyValue(params), colliders[i].units(), eps_units);
+        if (colliders[i].hasKey("rate-constants")) {
+            rateObjs.push_back(PlogRate(colliders[i], rate_units));
+            dataObjs.push_back(PlogData());
+            epsObjs1.push_back(epsObj_i);
+            epsObjs2.push_back(epsObj_i);
+        }
+        else if (colliders[i].hasKey("Troe")) {
+            // Value of "type" is unimportant; just needed to make falloff.cpp run
+            colliders[i]["type"];
+            rateObjs.push_back(TroeRate(colliders[i], rate_units));
+            dataObjs.push_back(FalloffData());
+            epsObjs1.push_back(epsObj_i);
+            epsObjs2.push_back(epsObj_i);
+        }
+        else if (colliders[i].hasKey("pressure-range")) {
+            rateObjs.push_back(ChebyshevRate(colliders[i], rate_units));
+            dataObjs.push_back(ChebyshevData());
+            epsObjs1.push_back(epsObj_i);
+            epsObjs2.push_back(epsObj_i);
+        }
+        // Collider has an eps specified, but no other info is provided. Assign it the same rate and data objects as "M"
+        else {
+            rateObjs.push_back(rateObj_M);
+            dataObjs.push_back(dataObj_M);
+            epsObjs1.push_back(epsObj_i);
+            epsObjs2.push_back(epsObj_M);
+        }
+    }
```

Comment:
Since this seems to be the most common case (based on the mechanism used in the examples), for the sake of efficiency I think you want to avoid recalculating the value of `rateObj_M` for each collider when (I think) it will just have the same value for all of them and all you really need is to compute it once and scale it by the sum of the efficiency-weighted mole fractions.

Reply:
This case actually only applies to colliders which have been explicitly specified with an eig0 or eps value but lack a k(TP) parameterization. For example, in the reaction H + O2 (+M) <=> HO2 (+M) this applies to only He, N2, H2, CO2, NH3, and H2O (thereby leaving 36 other species to be calculated later on in one step using the mole fraction sum). If Troe/PLOG/Chebyshev were provided for He, e.g., then it would be treated using the other branches of this conditional structure. Modifying this aspect of the code would make evalFromStruct() more complicated later on. I would prefer to leave it as-is, but am happy to modify it if you think the computational cost is too high.